bin/rollup.js

```diff
@@ -0,0 +1,138 @@
+ *
+ * Inspired by https://github.com/oat-sa/rollup-plugin-wildcard-external
+ */
+const wildcardExternalsPlugin = (peerDependencies) => ({
+    name: 'wildcard-externals',
+    resolveId(source, importer) {
+        if (importer) {
+            let matchesExternal = false;
+            peerDependencies.forEach((peerDependency) => {
+                if (source.includes(`/${peerDependency}/`)) {
+                    matchesExternal = true;
+                }
+            });
+
+            if (matchesExternal) {
+                return {
+                    id: source,
+                    external: true,
+                    moduleSideEffects: true,
+                };
+            }
+        }
+
+        return null; // other ids should be handled as usually
+    },
+});
+
+/**
+ * Moves the generated TypeScript declaration files to the correct location.
+ *
+ * This could probably be configured in the TypeScript plugin.
+ */
+const moveTypescriptDeclarationsPlugin = (packageRoot) => ({
+    name: 'move-ts-declarations',
+    writeBundle: async () => {
+        const isBridge = packageRoot.includes('src/Bridge');
+        const globPattern = path.join(packageRoot, 'dist', '**', 'assets', 'src', '**/*.d.ts');
+        const files = glob.sync(globPattern);
+
+        files.forEach((file) => {
+            // a bit odd, but remove first 8 or 11 directories, which will leave
+            // only the relative path to the file
+            const relativePath = file
+                .split('/')
+                .slice(isBridge ? 11 : 8)
+                .join('/');
+            const targetFile = relativePath.replace(
+                `${relativePath
+                    .split('/')
+                    .slice(1, isBridge ? 7 : 4)
+                    .join('/')}/`,
+                ''
+            );
+
+            if (!fs.existsSync(path.dirname(targetFile))) {
+                fs.mkdirSync(path.dirname(targetFile), { recursive: true });
+            }
+            fs.renameSync(file, targetFile);
+        });
+    },
+});
+
+/**
+ * @param {String} packageRoot
+ * @param {Array<String>} inputFiles
+ */
+function getRollupConfiguration({ packageRoot, inputFiles }) {
+    const packagePath = path.join(packageRoot, 'package.json');
+    const packageData = JSON.parse(fs.readFileSync(packagePath, 'utf8'));
+    const peerDependencies = [
+        '@hotwired/stimulus',
+        ...(packageData.peerDependencies ? Object.keys(packageData.peerDependencies) : []),
+    ];
+
+    inputFiles.forEach((file) => {
+        // custom handling for StimulusBundle
+        if (file.includes('StimulusBundle/assets/src/loader.ts')) {
+            peerDependencies.push('./controllers.js');
+        }
+
+        // React, Vue
+        if (file.includes('assets/src/loader.ts')) {
+            peerDependencies.push('./components.js');
+        }
+    });
+
+    const outDir = path.join(packageRoot, 'dist');
+
+    return {
+        input: inputFiles,
```

Comment:
We now spawn one `rollup` process per package, instead of one `rollup` process per package per files.